inc/inventory/asset/mainasset.class.php

```diff
@@ -364,17 +364,26 @@ public function prepareAllRulesInput(\stdClass $val): array {
          $input['tag'] = $this->getAgent()->fields['tag'];
       }
 
-      if (property_exists($val, 'serial') && !empty($val->serial)) {
-         $input['serial'] = $val->serial;
```

Comment:
serial/otherserial dropped ?

Reply:
Nope, all properies from `$val` are now used, including `serial` and `otherserial` if they're present.